runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -2906,7 +2906,7 @@ MM_CopyForwardScheme::updateScanStats(MM_EnvironmentVLHGC *env, J9Object *object
 		noEvacuation = isObjectInNoEvacuationRegions(env, objectPtr);
 	}
 
-	if ((_abortInProgress || noEvacuation) && isObjectInEvacuateMemory(objectPtr)) {
+	if ((_abortInProgress || noEvacuation || abortFlagRaised()) && isObjectInEvacuateMemory(objectPtr)) {
```

Comment:
I think it's safe to remove _abortInProgress (if it's true, then abortFlagRaised is true, too).

Not necessarily as part of this change, but perhaps we can remove costly isObjectInEvacuateMemory() check, if we check DIRTY_CARD reason first (if it's not DIRTY_CARD, it better be an object from collection set). Does that sound right?

Reply:
I agree that abortFlagRaised() does cover _abortInProgress==true cases, but  _abortFlag is volatile,  could checking _abortInProgress first save something?

Reply:
You mean performance wise? Well, volatile makes sure that variable is not register optimized, but I doubt _abortInProgress is. Even if it is,  abortFlagRaised() should not be much more expensive. It would be likely in CPU data cache if we are already here (doing some marking style of operation).